### PR TITLE
Added stats collection and reporting.

### PR DESCRIPTION
**Pull request by mollyling**
Also to consider: should we print stats to the webpage returned on the '/' handler?

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -32,4 +37,24 @@ io.on('connection', function(socket) {
   socket.on('disconnect', function() {
     disconnectMessages.forEach(doEmit);
   });
+  socket.on('connect_error', function(error) {
+      debug('ERROR: ' + JSON.stringify(error));
```
Comment (bemasc) [inline app.js]: Needs indentation help throughout

File: app.js [hunk 2]
```diff
@@ -32,4 +37,24 @@ io.on('connection', function(socket) {
...
+  });
 });
+
+function report() {
+    var end = process.hrtime();
+    if (stats.last_update) {
+        var delta_t_sec = (((end[0] - stats.last_update[0]) * 1000000000) +
```
Comment (bemasc) [inline app.js]: Prefer `1e9` for typo avoidance

File: app.js [hunk 3]
```diff
@@ -32,4 +37,24 @@ io.on('connection', function(socket) {
...
+            (end[1] - stats.last_update[1])) / 1000000000;
```
Comment (bemasc) [inline app.js]: How about
`end[0] - stats.last_update[0] + (end[1] - stats.last_update[1]) / 1e9`?

File: package.json [hunk 4]
```diff
@@ -12,6 +12,9 @@
   "license": "Apache Version 2.0",
   "dependencies": {
     "express": "^4.12.0",
-    "socket.io": "~1.0"
+    "socket.io": "~1.0",
```
Comment (bemasc) [inline package.json]: Should these be "~" or "^"?

Comment (dborkan) [inline package.json]: Lets make these `"socket.io": "^1.2"`, same as the client https://github.com/uProxy/freedom-social-quiver/blob/master/package.json#L43

File: app.js (hunk 3 above)
Comment (mollyling) [inline app.js]: That's fine.  I added another parens around the entire division, to make that part a little clearer.

**Comment by mollyling** [comment]
PTAL, thanks.

File: app.js [hunk 5]
```diff
@@ -2,6 +2,10 @@
 var app = require('express')();
 var server = require('http').Server(app);
 var io = require('socket.io')(server);
+var process = require('process');
+var debug = require('debug')('stats');
+var stats = { 'messages' : 0, 'last_update' : null };
```
Comment (bemasc) [inline app.js]: Let's keep all the requires in one block and move `var stats` below it.

File: app.js [hunk 6]
```diff
@@ -45,3 +50,20 @@ io.on('connection', function(socket) {
     console.log('Error:', err);
   });
 });
+
+function report() {
+  var end = process.hrtime();
+  if (stats.last_update) {
+    var delta_t_sec = end[0] - stats.last_update[0] +
+      ((end[1] - stats.last_update[1]) / 1e9);
+    debug("clients: " + io.engine.clientsCount + ", message rate: " +
```
Comment (bemasc) [inline app.js]: Let's standardize on single-quotes for strings.

File: app.js [hunk 7]
```diff
@@ -45,3 +50,20 @@ io.on('connection', function(socket) {
     console.log('Error:', err);
   });
 });
+
+function report() {
+  var end = process.hrtime();
+  if (stats.last_update) {
+    var delta_t_sec = end[0] - stats.last_update[0] +
+      ((end[1] - stats.last_update[1]) / 1e9);
```
Comment (bemasc) [inline app.js]: 4-space indent

**Comment by bemasc** [comment]
:+1: with nits

**Pull request merged**